### PR TITLE
keyword-only args for chat dataset, optional chat format

### PR DESCRIPTION
**Pull request by ebsmothers**
Not all datasets need chat format, so we should make it optional. Also we should be using keyword-only args more. In this case the latter is beneficial for the former, so enable it for ChatDataset.

#### Test plan:

```
pytest tests
...
======= 191 passed, 19 skipped, 4 warnings in 60.28s (0:01:00) =========
```

```
pytest tests -m integration_test
...
======= 14 passed, 4 skipped, 192 deselected, 2 warnings in 190.73s (0:03:10) =========
```

Now we can run without applying chat format via e.g. 

```
tune run lora_finetune_single_device --config llama3/8B_lora_single_device dataset=torchtune.datasets.slimorca_dataset dataset.chat_format=null
...
1|32|Loss: 2.4104795455932617:   0%|                                    
```

### CODE REVIEW

**Review by joecummings — APPROVED** [review]

File: torchtune/datasets/_chat.py [hunk 1]
```diff
@@ -50,19 +50,21 @@ class ChatDataset(Dataset):
             (https://huggingface.co/docs/datasets/en/package_reference/loading_methods#datasets.load_dataset.path)
         convert_to_messages (Callable[[Mapping[str, Any]], List[Message]]): function that keys into the desired field in the sample
             and converts to a list of `Messages` that follows the llama format with the expected keys
-        chat_format (ChatFormat): template used to format the chat. If the placeholder variable
+        chat_format (Optional[ChatFormat]): template used to format the chat. If the placeholder variable
```
Comment (RdoubleA) [inline torchtune/datasets/_chat.py:53]: let me know if this is too much details to add
```suggestion
        chat_format (Optional[ChatFormat]): template used to format the system, user, and assistant messages in the chat. This is used to add structured text around the actual messages, such as the [INST] tags in LLaMA2 and in Mistral. The extra text will still get tokenized as normal text, not as special tokens with separate embedding vectors. In models like LLaMA3 where the tokenizer adds tags as special tokens, `chat_format` is not needed, unless you want to structure messages in a particular way for inference. If the placeholder variable
```

File: torchtune/datasets/_slimorca.py [hunk 2]
```diff
@@ -4,16 +4,20 @@
...
 from torchtune.datasets._chat import ChatDataset
 
 from torchtune.modules.tokenizers import Tokenizer
 
 
 def slimorca_dataset(
+    *,
     tokenizer: Tokenizer,
     source: str = "Open-Orca/SlimOrca-Dedup",
+    chat_format: Optional[ChatFormat] = Llama2ChatFormat,
```
Comment (RdoubleA) [inline torchtune/datasets/_slimorca.py:20]: should this still be a required parameter? this is not easy to configure from the config (without a nested component)

File: torchtune/datasets/_chat.py (hunk 1 above)
Comment (ebsmothers) [reply torchtune/datasets/_chat.py:53]: It is a bit long, but I think it's useful. Gonna mostly adopt but will leave out the bit about separate embedding vectors

File: torchtune/datasets/_slimorca.py (hunk 2 above)
Comment (ebsmothers) [reply torchtune/datasets/_slimorca.py:20]: Yeah it's not perfect but see my test command, now I can at least disable to run Llama3 without the chat format from CLI alone

**Pull request merged**